### DISCUSSION TITLE
Better type-support for the `value` from the default form slot

### DISCUSSION THREAD

**Discussion by gbyesiltas** [formkit/formkit#913] [Ideas]
<img width="821" alt="Screenshot 2023-08-24 at 17 20 17" src="https://github.com/formkit/formkit/assets/50772478/05cce9e1-ba76-41c8-bb8a-519497c63bf9">

Would be awesome if the type of the form value could be inferred with some generic component magic 🥹

**Comment by justin-schroeder** [comment]
This would be awesome, but unfortunately there isn’t enough data to make this reliable unless you always specified a prop with the entire structure of your form.
